benchmarks/src/compare.ts

```diff
@@ -0,0 +1,244 @@
+        (collected[key] = {
+          mode,
+          browser,
+          n: parseInt(n),
+          type,
+          times: { old: [], new_: [] }
+        });
+      data.times[version].push(parseFloat(time));
+    }
+  }
+  for (const {
+    mode,
+    browser,
+    type,
+    n,
+    times: { old, new_ }
+  } of Object.values(collected)) {
+    if (old.length != new_.length) {
+      console.warn('Skipping because different lengths between runs', {
+        mode,
+        browser,
+        type,
+        n
+      });
+      continue;
+    }
+    yield {
+      mode,
+      browser,
+      type,
+      n,
+      ...performanceChangeFromData(old, new_, confidenceInterval)
+    };
+  }
+}
+
+/**
+ * Quantifies the performance changes between two measures systems. Assumes we gathered
+ * n independent measurement from each, and calculated their means and varience.
+ *
+ * Based on the work by Tomas Kalibera and Richard Jones. See their paper
+ * "Quantifying Performance Changes with Effect Size Confidence Intervals", section 6.2,
+ * formula "Quantifying Performance Change".
+ *
+ * However, it simplifies it to only assume one level of benchmarks, not multiple levels.
+ * If you do have multiple levels, simply use the mean of the lower levels as your data,
+ * like they do in the paper.
+ *
+ * @param oldSystem The old system we measured
+ * @param newSystem The new system we measured
+ * @param n The number of samples from each system (must be equal)
+ * @param confidenceInterval The confidence interval for the results.
+ *  The default is a 95% confidence interval (95% of the time the true mean will be
+ *  between the resulting mean +- the resulting CI)
+ */
+export function performanceChange(
+  { mean: y_o, variance: s_o }: { mean: number; variance: number },
+  { mean: y_n, variance: s_n }: { mean: number; variance: number },
+  n: number,
+  confidenceInterval: number = 0.95
+): { mean: number; confidenceInterval: number } {
+  const dof = n - 1;
+  const t = quantile(1 - (1 - confidenceInterval) / 2, dof);
+  const oldFactor = sq(y_o) - (sq(t) * s_o) / n;
+  const newFactor = sq(y_n) - (sq(t) * s_n) / n;
+  const meanNum = y_o * y_n;
+  const ciNum = Math.sqrt(sq(y_o * y_n) - newFactor * oldFactor);
+  return {
+    mean: meanNum / oldFactor,
+    confidenceInterval: ciNum / oldFactor
+  };
+}
+
+/**
+ * Compute the performance change based on a number of old and new measurements.
+ */
+export function performanceChangeFromData(
+  old: number[],
+  new_: number[],
+  confidenceInterval: number = 0.95
+): { mean: number; confidenceInterval: number } {
+  const n = old.length;
+  if (n !== new_.length) {
+    throw new Error('Data have different length');
+  }
+  return performanceChange(
+    { mean: mean(...old), variance: variance(...old) },
+    { mean: mean(...new_), variance: variance(...new_) },
+    n,
+    confidenceInterval
+  );
+}
+
+/**
+ * Format a performance changes like `10∓0.3`
+ */
+function formatChange({
+  mean,
```

Comment:
'formatChange' is defined but never used - This is what GitHub Actions / Linux (integrity, 3.8) says...